doc/Data Plane Code Generation/ServiceDrivenEvolution.md

```diff
@@ -0,0 +1,191 @@
+{
+    ...
+    try
+    {
+        using HttpMessage message = CreatePostParametersRequest(content, ContentType.ApplicationJson, context);
+        return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+    }
+    ...
+}
+```
+
+**Notice:**
+1. We leverage the `CreatePostParametersRequest` method in V2 but always pass in the `ContentType.ApplicationJson` as the `contentType` parameter in the manual method to make the behavior align with V1.
+2. Whether to add the attribute `[EditorBrowsable(EditorBrowsableState.Never)]` depends on whether you want to discourage callers from using this. E.g., if service add a new ContentType value to improve performance, you should add this attribute so new users wouldn't discover it, while if service has default preference of old ContentType value, you should not add this attribute to make it easy for users to discover the default.
+
+## A method changes a required parameter to optional
+
+Generated code in a V1 client with two required parameters:
+
+``` C#
+public virtual async Task<Response> PutRequiredOptionalAsync(string requiredParam1, string requiredParam2, RequestContext context = null)
+{
+    ...
+    try
+    {
+        using HttpMessage message = CreatePutRequiredOptionalRequest(requiredParam1, requiredParam2, context);
+        return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+    }
+    ...
+}
+```
+
+**Required input is changed to an optional input when it is in the last position (i.e., `requiredParam2`).**
+
+This case is safe to generate as:
+
+``` C#
+public virtual async Task<Response> PutRequiredOptionalAsync(string requiredParam1, string requiredParam2 = null, RequestContext context = null)
+{
+    ...
+    try
+    {
+        using HttpMessage message = CreatePutRequiredOptionalRequest(requiredParam1, requiredParam2, context);
+        return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+    }
+    ...
+}
+```
+
+**Required input is changed to an optional input when it is not in the last position (e.g., `requiredParam2`).**
+
+In this case, below generated code is not safe to go, which would end up passing the wrong values to the method parameters, without any warnings from the compiler.
+``` C#
+public virtual async Task<Response> PutRequiredOptionalAsync(string requiredParam2, string requiredParam1 = null, RequestContext context = null)
+{
+    ...
+    try
+    {
+        using HttpMessage message = CreatePutRequiredOptionalRequest(requiredParam2, requiredParam1, context);
+        return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+    }
+    ...
+}
+```
+You have to handle this via a manually-written convenience overload. E.g.,
+``` C#
+public virtual async Task<Response> PutRequiredOptionalAsync(string requiredParam2, string requiredParam1 = null, RequestContext context = null)
```

Comment:
For this part, i don't quite get it.  It seems the api signature is the same, but you change the parameter order inside  CreatePutRequiredOptionalRequest.   Will the latter one replace the generated code?